src/security/ErrorDetail.cc

```diff
@@ -716,18 +724,20 @@ Security::ErrorDetail::err_lib_error() const
 size_t
 Security::ErrorDetail::convert(const char *code, const char **value) const
 {
+    // TODO change this to return SBuf
```

Comment:
There are a lot of problems in this official code, and these problems are in the way of doing the Right Thing in this PR. I am too busy to fix this quickly myself. If you can help, I can suggest the next step: This method should take an std::ostream reference (instead of a pointer to a c-string). The error extractors in the map below should print their information into the supplied stream (instead of returning a string).

Reply:
> This method should take an std::ostream reference (instead of a pointer to a c-string). The error extractors in the map below should print their information into the supplied stream (instead of returning a string).

Done in 326b13c, but I have not reviewed, tested, and polished the result yet.

Reply:
> Done in [326b13c](https://github.com/squid-cache/squid/commit/326b13c98494cca1d95241d052f6db7bf425f495), but I have not reviewed, tested, and polished the result yet.

I pushed more fixes (as of commit 51ba982), but I am not done yet.